### PR TITLE
Added badges for skin

### PR DESCRIPTION
**Pull request by agliga**
Spec is according to playbook badges.

## Description

Added css classes for badges. One style is a normal badge that follows text. The second one is a style that is shown on the upper right side of an icon.

## Context

I set the font-size to em in order for it to be relative size to the parent. However, there are no em variables in DS4/6 so I added my own one.
I also set all paddings and margins to pixels as per Ian.

### For inline badge:
Initially I tried to make the badge a span element. It worked fine until the digit inside the span was slightly smaller (ie the space a 1 takes is less than 4). So I had to switch the display type from inline to either inline-block or inline-flex. I went with flex because with flex I could use justify-content and align-items to center the digit inside (yes it was not aligning vertically correct without this). I could use line-height + text-align but justify-content and align-items was better for getting it pixel perfect.
I set a high in order for the badge not to fill in the menu item (as inline it was fine, but when it became block it filled in horizontally because any child of menu has a flex of 1)

### For floating badge on icon:
I added a parent container class to the "icon" class. Initially I made it only add a position relative and then I would control the position using a class to place it either on right/left/top/bottom. However, the spec only has top right, so I changed the "container" class to apply all the positioning styles to place the badge on the top right of the icon. 

## References

EBAYUI-33 Jira Issue

### DS4-chrome
ds4-chrome

### DS6-chrome
ds6-chrome

### DS4-firefox
ds4-firefox

### DS6-firefox
ds6-firefox

### DS4-safari
ds4-safari

### DS6-safari
ds6-safari

### CODE REVIEW

**Comment by seangates** [comment]
Looks pretty good 200% zoomed on Chrome and Safari:

image

Looks like Firefox it breaks down a little:

image

@ianmcburnie Is it proper to test 200% zooming on Firefox by updating the default root font size to 32px?

**Comment by seangates** [comment]
Here's the page zoom on FF at 200% (as opposed to font changes):

image

**Comment by seangates** [comment]
@agliga Overall, looks awesome!

**Comment by seangates** [comment]
Is there a reason we shouldn't use `box-sizing: border-box;` in order to set `min-width` and `min-height` to 24px? (20px width + 2x2px borders)

File: src/less/badge/ds4/badge.less [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@import '../base/badge.less';
+@import "../../less/ds4/variables.less";
+@import "../../less/ds4/mixins.less";
+
+.badge {
+    background: @ds4-color-core-red;
+    color: @ds4-color-core-white;
+
+    &-container .badge {
+        border: 2px solid @ds4-color-core-white;
```
Comment (seangates) [inline src/less/badge/ds4/badge.less]: Let's add `border: 2px solid;` to the `/badge/base/badge.less` file, and just do `border-color: @ds4-color-core-white;` here. That makes it explicit we're only setting the color here, and not the entire border rule.

File: src/less/badge/ds6/badge.less [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@import '../base/badge.less';
+@import "../../less/ds6/variables.less";
+@import "../../less/ds6/mixins.less";
+
+.badge {
+    background-color: @ds6-color-badge-background;
+    color: @ds6-color-badge-text;
+
+    &-container .badge {
+        border: 2px solid @ds6-color-badge-text;
```
Comment (seangates) [inline src/less/badge/ds6/badge.less]: Let's add `border: 2px solid;` to the `/badge/base/badge.less` file, and just do `border-color: @ds6-color-badge-text;` here. That makes it explicit we're only setting the color here, and not the entire border rule.

**Comment by ianmcburnie** [comment]
@agliga If I change the icon badge value to 44, I see the following alignment issue:

screen shot 2018-10-03 at 3 01 12 pm

File: dist/badge/ds4/badge.css [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  position: absolute;
+  top: -8px;
+  z-index: 1;
+}
+.icon-badge-container {
+  position: relative;
+}
+.text-badge,
+.icon-badge {
+  background: #dd1e31;
```
Comment (ianmcburnie) [inline dist/badge/ds4/badge.css]: Minor, but should be `background-color` I believe?

File: docs/_includes/common/badge.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
+<div id="badge">
+    <h2><span class="secondary-text">@ebay/skin/</span>badge</h2>
+    <p>Badges are normally used to display an indication to the user about some important notification information.</p>
+
+    <p>Badges can be displayed individually</p>
+    <div class="demo">
+        <p>
```
Comment (ianmcburnie) [inline docs/_includes/common/badge.html]: This example is missing a 'demo__inner' which is causing it to look a bit odd.

**Comment by agliga** [comment]
@ianmcburnie I tested on achor tags. Looks fine!
image

I also added the anchor tags in the test files.

**Comment by seangates** [comment]
Yes, great work, @agliga!

File: docs/static/ds6/docs.js [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
-$_mod.installed("makeup-expander$0.5.0","makeup-focusables","0.0.3");
```
Comment (ianmcburnie) [inline docs/static/ds6/docs.js]: Oops, looks like this docs change sneaked in and reverted us back to older plugin versions. I will fix directly in 6.1.0 branch.

**Pull request merged**